math/genvector/inc/Math/GenVector/Plane3D.h

```diff
@@ -230,16 +250,71 @@ namespace Math {
       /**
          Normalize the normal (a,b,c) plane components
       */
-      void Normalize();
+      template< typename SCALAR = T >
+      typename std::enable_if< std::is_arithmetic<SCALAR>::value, void >::type
+      Normalize()
+      {
+        // normalize the plane
+        const SCALAR s = std::sqrt( fA*fA + fB*fB + fC*fC );
```

Comment:
Yes, because this was a case where writing generic code was not easy, due to the bool/mask differences. I decided to instead specialise with type traits. If you check you will see there are two implementations, for the scalar and vector types.